File: Linnworks/src/php/Factory.php
<?php
class Factory
{
    public static function GetResponse($ext, $data, $token, $server)
    {
        $opts = array(
		  'http'=>array(
			'method'=>"POST",
			'header'=>"Content-Type: application/x-www-form-urlencoded; charset=UTF-8\r\n" .
					  "User-Agent: Linnworks PHP API SDK \r\n" . 
					  "Referer: https://www.linnworks.net/ \r\n" .
					  "Content-Length: " . strlen($data) . "\r\n" .
					  "Authorization: " . $token . "\r\n"
					  ,
			'content' => $data
		  )
		);
		
		$context = stream_context_create($opts);
		
		$url = "";
				
		if(!empty($token))
		{
			$url = $server . '/api/' . $ext;
		} else {
			$url = 'https://api.linnworks.net/api/' . $ext;
		}
		
		$response = file_get_contents($url, FALSE, $context);
		
		return $response;
    }
}
?>

File: Linnworks/src/netcore/LinnworksAPI/Factory.cs
using System;
using System.IO;
using System.Net;
using System.Text;

namespace LinnworksAPI
{
    public class Factory
    {
        public static string GetResponse(string extension, string body, ApiContext context, string httpMethod, int? timeout)
        {
            string url = context.ApiServer + extension;

            bool isGet = httpMethod == "GET";

            if (!string.IsNullOrWhiteSpace(body) && isGet)
            {
                url += "?" + body;
            }

            var req = HttpWebRequest.Create(url);
            req.Method = httpMethod;

            if (!isGet)
            {
                req.ContentType = "application/x-www-form-urlencoded";
            }

            req.Headers.Add("RecursionCount", context.RecursionCount.ToString());

            if (timeout.HasValue)
                req.Timeout = timeout.Value;

            if (context.SessionId != Guid.Empty)
                req.Headers.Add(HttpRequestHeader.Authorization, context.SessionId.ToString());

            if (!string.IsNullOrWhiteSpace(body) && !isGet)
                req.ContentLength = Encoding.UTF8.GetBytes(body).Length;

            if (!string.IsNullOrWhiteSpace(body) && !isGet)
            {
                using (Stream post = req.GetRequestStream())
                {
                    using (StreamWriter writer = new StreamWriter(post))
                    {
                        writer.Write(body);
                    }
                }
            }

            var response = (HttpWebResponse)req.GetResponse();
            string responseBody = "";

            using (StreamReader sr = new StreamReader(response.GetResponseStream()))
            {
                responseBody = sr.ReadToEnd();
            }

            if ((int)response.StatusCode < 200 || (int)response.StatusCode > 299)
            {
                var error = JsonFormatter.ConvertFromJson<ApiError>(responseBody);
                throw new Exception(error.Message);
            }

            return responseBody;
        }

        public class ApiError
        {
            public string Message { get; set; }
        }
    }
}
